### ISSUE TITLE
Adding Threshold Trigger - Channel 'Total' is not a valid

### ISSUE THREAD

**Issue by Eazhi** [lordmilko/PrtgAPI#27]
Hi @lordmilko !

I'm having trouble with the Trigger creation. The trigger I'm trying to add is a `Threshold Trigger` and I am trying to add it to a `Windows CPU Load` sensor.

At first I tried this :
```C#
var p = new ThresholdTriggerParameters(3038)
{
    Channel = TriggerChannel.Total,
    Condition = TriggerCondition.Above,
    Latency = 120,
    OnNotificationAction = client.GetNotificationActions(Property.Name, "Email to Admin")[0],
    OffNotificationAction = client.GetNotificationActions(Property.Name, "Email to Admin")[0],
    Threshold = 80,
};
var trigger = client.AddNotificationTrigger(p);
```
This gives me back : 
`System.InvalidOperationException: Channel 'Total' is not a valid value for sensor with ID 3038. Triggers assigned directly to sensors must refer to a specific Channel or Channel ID.`

Using `TriggerChannel.Primary` gives back the same error.

Which is weird because when I set Channel to something that shouldn't work:
Same code but with : `Channel = TriggerChannel.TrafficIn,`
It gives me back `System.InvalidOperationException: Only 'Primary', 'Total' and sensor specific channels are valid for threshold triggers`

I'm also not sure that this is the right way to set On/OffNotificationAction, I've tried without them (to leave them null) but I had the same errors, so it didn't seem to be the problem here.

So I tried it another way, in the Trigger Channels part of the documentation I found this :
> For notification triggers that are applied directly to sensors, a Channel object must be specified, which can be achieved via an implicit cast

Therefore, I tried :
```C#
var channel = client.GetChannels(3038, "Total")[0]; // Gives back the Total Channel
var p = new ThresholdTriggerParameters(3038)
{
    Channel = channel,
    Condition = TriggerCondition.Above,
    Latency = 120,
    OnNotificationAction = client.GetNotificationActions(Property.Name, "Email to Admin")[0],
    OffNotificationAction = client.GetNotificationActions(Property.Name, "Email to Admin")[0],
    Threshold = 80,
};
var trigger = client.AddNotificationTrigger(p);
```

This still gives me the same error `System.InvalidOperationException: Channel 'Total' is not a valid value for sensor with ID 3038. Triggers assigned directly to sensors must refer to a specific Channel or Channel ID.`

Since these didn't work I tried creating the new Trigger from an existing one like this :
```C#
var trigger = client.GetNotificationTriggers(6887)[0]; // Gives back a threshold trigger
var p = new ThresholdTriggerParameters(3038, trigger);
var newTrigger = client.AddNotificationTrigger(p);
```
This also gives me back the same error `System.InvalidOperationException: Channel 'Total' is not a valid value for sensor with ID 3038. Triggers assigned directly to sensors must refer to a specific Channel or Channel ID.`

Doing it by hand on PRTG works fine, so I don't think that I passed any wrong parameters.

Thanks for this API and for taking the time to answer !

**Comment by lordmilko** [comment]
Hi @Eazhi,

I can confirm that there are two issues going on here
1. `TriggerChannel.ParseInternal` does not correctly stipulate that `TriggerChannel` objects containing a `Channel` should not be try to be parsed to a `StandardTriggerChannel`. As a result, PrtgAPI erroneously converts your `Channel` object into a `StandardTriggerChannel` enum member. This is causing the first issue you are seeing
2. When `PrtgClient.UpdateTriggerChannels` attempts to resolve the name of the channel returned by PRTG to either a `Channel` or a `StandardTriggerChannel` (Primary / Total / TrafficIn / TrafficOut), it does not consider the possibility that you could actually have a channel *called "Total"!* As a result, the `NotificationTrigger` object (and any `TriggerParameters` created from it) end up with a Channel property containing enum member `StandardTriggerChannel.Total`, preventing the trigger from being added to another object without first updating the parameter object's Channel property

To work around this issue for now, you can manually create a `TriggerChannel` object containing the *channel ID* of the channel you wish to use on the notification trigger

```c#
var channel = client.GetChannels(3038, "Total")[0]; // Gives back the Total Channel
var action = client.GetNotificationActions(Property.Name, "Email to Admin")[0];

var p = new ThresholdTriggerParameters(3038)
{
    Channel = new TriggerChannel(channel.Id),
    Condition = TriggerCondition.Above,
    Latency = 120,
    OnNotificationAction = action,
    OffNotificationAction = action,
    Threshold = 80,
};

var trigger = client.AddNotificationTrigger(p);
```

Are you able to confirm whether this workaround worked for you?

Regards,
lordmilko

**Comment by Eazhi** [comment]
That worked perfectly!
 
Thanks for answering so fast!

**Comment by lordmilko** [comment]
Hi @Eazhi,

Good to hear! FYI this issue has been fixed in PrtgAPI 0.9.0, which is currently due to be released in about a month or two.

Please let me know if you have any further issues

Regards,
lordmilko

**Comment by lordmilko** [comment]
Hi @Eazhi,

PrtgAPI 0.9.0 has now been released! A number of bugs surrounding the use of trigger channels were identified and resolved as a result of investigating this issue. Thanks!

To update to the latest version of PrtgAPI, simply run

```powershell
Update-Package PrtgAPI
```

from within the NuGet Package Manager console in Visual Studio. Or, alternatively you can upgrade via the NuGet package browser. Please be aware of any [breaking changes](https://github.com/lordmilko/PrtgAPI/releases/tag/v0.9.0) that may affect you before considering upgrading.

Please let me know if you have any issues

Regards,
lordmilko
